reconcile/utils/gql.py

```diff
@@ -198,13 +193,39 @@ def get_resource(path: str) -> dict[str, Any]:
     return get_api().get_resource(path)
 
 
+class PersistentRequestsHTTPTransport(RequestsHTTPTransport):
+    def connect(self):
+        if self.session is None:
+            # Copied over from RequestsHTTPTransport
+            self.session = requests.Session()
+            # we did not implement this in our copy!
+            assert self.retries == 0
+
+    def close(self) -> None:
+        pass
+
+
 def _init_gql_client(url: str, token: Optional[str]) -> Client:
+    global _local_client
```

Comment:
why is this global if you want to define per thread?

Reply:
That question is spot on with the problems in this code. The issue here is, that the GraphQL API instance is a singleton and not created per thread (which would be the best solution IMHO but would require some (I think a lot) of refactoring). `_local_client` must be global, cause a threading.local instance must be initialized only once and then shared by the threads.

Reply:
Ok. Since you're overriding the `connect` method, you will get rid of the `TransportAlreadyConnected` since you actually want to share the connection between threads... But can it cause problems if two threads need to use the connection at the same time?

Reply:
That's the trickery of threading.local. It provides a data structure, where threads don't share data. In the end this means every thread has it's own Client instance

Reply:
cool!

I guess all this deserves a nice comment in the code for the sake of the future readers :) I'm sure they will have similar doubts 😅

Reply:
Added a comment!

Reply:
Added a comment!